packages/compiler/src/render3/partial/api.ts

```diff
@@ -232,6 +232,14 @@ export interface R3DeclareQueryMetadata {
    */
   descendants?: boolean;
 
+  /**
+   * True to only fire changes if there are underlying changes to the query. This will become `true`
+   * be default in v12. `QueryList.changes` would fire even if no changes to the query list were
+   * detected. This is not ideal, as changes should only fire if the `QueryList` actually materially
+   * changed.
+   */
+  emitDistinctChangesOnly?: boolean;
```

Comment:
```suggestion
  emitDistinctChangesOnly: boolean;
```

In the compiler we try to avoid optional fields, especially optional `boolean`s. (I know `descendants` above is in violation of this rule - it should really be non-optional too). Prefer explicitly passing the property.

Reply:
This one is special ;) This interface documents the structure as generated by the linker, where we omit properties when they are `false` (or another typical default).

Reply:
That brings up an interesting question though. If the plan is to change this property from default-false to default-true in a future version, how will that affect prelinked code already on NPM (hypothetically assuming such code exists)? Should we preemptively always emit the property and set it explicitly to `false`, with the expectation that a missing value will eventually be interpreted as `true`? Or do we have some other way of changing the default without affecting existing prelinked code?

Reply:
We discussed this a while ago but I don't recall the exact outcome. But yes, the linker's defaults are never meant to be switched, as that would affect published code.

Yesterday we agreed to start a document with public API considerations, I'll incorporate this aspect as well so we can reassess (the plan is to present that document at next week's sync)

Reply:
Ty! Looking forward to it. This code LGTM for now